src/olympia/discovery/tests/test_views.py

```diff
@@ -64,65 +26,61 @@ def _check_disco_addon(self, result, item, flat_name=False,
-            assert '</span>' in result['heading']
-        else:
-            assert u'{1} <span>by <a href="{0}">{2}</a></span>'.format(
-                absolutify(addon.get_url_path()),
-                unicode(item.addon_name or addon.name),
-                u', '.join(author.name for author in addon.listed_authors)
-            ) == result['heading']
-        assert result['description']
-        assert len(result['addon']['previews']) == len(addon.current_previews)
+        assert result['heading'] == item.heading
+        assert result['description'] == item.description
 
     def _check_disco_theme(self, result, item, flat_name=False):
-        addon = self.addons[item.addon_id]
-        assert result['addon']['type'] == 'persona'
+        addon = item.addon
         assert result['addon']['id'] == item.addon_id == addon.pk
         if flat_name:
             assert result['addon']['name'] == unicode(addon.name)
         else:
             assert result['addon']['name'] == {'en-US': unicode(addon.name)}
         assert result['addon']['slug'] == addon.slug
-
-        assert u'{1} <span>by <a href="{0}">{2}</a></span>'.format(
-            absolutify(addon.get_url_path()),
-            unicode(item.addon_name or addon.name),
-            u', '.join(author.name for author in addon.listed_authors)
-        ) == result['heading']
-        description_output = (
-            (u'<blockquote>%s</blockquote>' % addon.description)
-            if addon.description else None)
-        assert result['description'] == description_output
         assert result['addon']['theme_data'] == addon.persona.theme_data
 
+        assert result['heading'] == item.heading
+        assert result['description'] == item.description
+
 
 class TestDiscoveryViewList(DiscoveryTestMixin, TestCase):
     def setUp(self):
         super(TestDiscoveryViewList, self).setUp()
         self.url = reverse_ns('discovery-list')
+        self.addons = []
 
-        self.addons = get_dummy_addons()
+        # This one should not appear anywhere, position isn't set.
+        DiscoveryItem.objects.create(addon=addon_factory())
+
+        for i in range(1, 8):
+            if i % 3:
+                type_ = amo.ADDON_PERSONA
+            else:
+                type_ = amo.ADDON_EXTENSION
+            addon = addon_factory(type=type_)
+            self.addons.append(addon)
+            DiscoveryItem.objects.create(addon=addon, position=i)
+
+        for i in range(1, 8):
+            if i % 3:
+                type_ = amo.ADDON_PERSONA
+            else:
+                type_ = amo.ADDON_EXTENSION
+            addon = addon_factory(type=type_)
+            DiscoveryItem.objects.create(addon=addon, position_china=i)
 
     def test_reverse(self):
         assert self.url.endswith(
             '/api/%s/discovery/' % api_settings.DEFAULT_VERSION)
 
     def test_list(self):
-        response = self.client.get(self.url, {'lang': 'en-US'})
+        with self.assertNumQueries(26):
```

Comment:
We can look into optimizing this, but the current disco pane is not that much better actually (around ~20 queries as well). It comes from the fact that we fetch a bunch of things for each add-on (categories, current version, applications versions, files, personas for themes, authors, previews, translations, translations of version fields) and not everything benefits from the Addon transformer.

Reply:
I'll push a fix that gets us down to 12 queries when removing extraneous queries for add-ons with no listed authors (so the number was artificially higher just because of bad test data)